### PR TITLE
story_AT1-215: Add exception handling. Improve documentation.

### PR DESCRIPTION
**Pull request by adityadangeska**
Added specific exception handling and improved documentation.

### CODE REVIEW

File: skabase/SKALogger/SKALogger.py [hunk 1]
```diff
@@ -190,12 +190,16 @@ def SetCentralLoggingLevel(self, argin):
...
-            i += 1
+            try:
+                self.info_stream("Central Logging level : %s, Device : %s",
+                                 central_logging_level[i],
+                                 central_logging_level[i])
+                dev_proxy = DeviceProxy(central_logging_device[i])
+                dev_proxy.centralLoggingLevel = central_logging_level[i]
+                i += 1
+            except DevFailed:
+                self.error_stream("Failed to set Central Logging level for [%s]", central_logging_device[i])
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: Wouldn't it be nice to include the original exception's error message in the log?

File: skabase/SKALogger/SKALogger.py [hunk 2]
```diff
@@ -190,12 +190,16 @@ def SetCentralLoggingLevel(self, argin):
...
-            dev_proxy = DeviceProxy(central_logging_device[i])
-            dev_proxy.centralLoggingLevel = central_logging_level[i]
-            i += 1
+            try:
+                self.info_stream("Central Logging level : %s, Device : %s",
+                                 central_logging_level[i],
+                                 central_logging_level[i])
+                dev_proxy = DeviceProxy(central_logging_device[i])
+                dev_proxy.centralLoggingLevel = central_logging_level[i]
+                i += 1
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: Having this inside the try..except, combined with not re-raising an exception on `DevFailed`, results in an infinite loop if we do get `DevFailed`.

File: skabase/SKALogger/SKALogger.py [hunk 3]
```diff
@@ -224,12 +228,15 @@ def SetElementLoggingLevel(self, argin):
...
-            dev_proxy = DeviceProxy(element_logging_device[i])
-            dev_proxy.elementLoggingLevel = element_logging_level[i]
-            i += 1
+            try:
+                self.info_stream("Element Logging level : %s, Device : %s",
+                                 element_logging_level[i],
+                                 element_logging_device[i])
+                dev_proxy = DeviceProxy(element_logging_device[i])
+                dev_proxy.elementLoggingLevel = element_logging_level[i]
+                i += 1
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: Same comments as above.

File: skabase/SKALogger/SKALogger.py [hunk 4]
```diff
@@ -258,12 +265,15 @@ def SetStorageLoggingLevel(self, argin):
...
-            dev_proxy = DeviceProxy(storage_logging_device[i])
-            dev_proxy.storageLoggingLevel = storage_logging_level[i]
-            i += 1
+            try:
+                self.info_stream("Storage logging level : %s, Device : %s",
+                                 storage_logging_level[i],
+                                 storage_logging_device[i])
+                dev_proxy = DeviceProxy(storage_logging_device[i])
+                dev_proxy.storageLoggingLevel = storage_logging_level[i]
+                i += 1
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: Same comments as above.

File: skabase/SKAMaster/SKAMaster.py [hunk 5]
```diff
@@ -178,10 +178,11 @@ def isCapabilityAchievable(self, argin):
         # PROTECTED REGION ID(SKAMaster.isCapabilityAchievable) ENABLED START #
         """
         Checks of provided capabilities can be achieved by the resource(s).
-        :param argin: DevVarLongStringArray. An array consisting pair of DevLong and
-        DevString. A single entry consists of\n
-            [nrInstances]: DevLong. Number of instances of the capability
-            [Capability types]: DevString. Type of capability
+
+        :parameter argin: DevVarLongStringArray. An array consisting pair of
```
Comment (ajoubertza) [inline skabase/SKAMaster/SKAMaster.py]: In the other files modified in this PR, the docstrings are using `:param`, not `:parameter`.  Those other files also have an extra blank line before describing the parameter.

What standard are we following for docstrings?

File: skabase/SKATelState/SKATelState.py [hunk 6]
```diff
@@ -92,8 +92,11 @@ def main(args=None, **kwargs):
     # PROTECTED REGION ID(SKATelState.main) ENABLED START #
     """
     Main function of the module
+
     :param args: None
```
Comment (ajoubertza) [inline skabase/SKATelState/SKATelState.py]: `None` is the default, not the type of the `args`, so this docstring is a bit confusing.

**Review by ajoubertza — CHANGES_REQUESTED** [review]

File: skabase/SKATelState/SKATelState.py (hunk 6 above)
Comment (adityadangeska) [reply skabase/SKATelState/SKATelState.py]: Updated.

File: skabase/SKAMaster/SKAMaster.py (hunk 5 above)
Comment (adityadangeska) [reply skabase/SKAMaster/SKAMaster.py]: Modified the documentation. 
We are using Sphinx style docstrings in general. Although we know the docstrings can be improved further, both in terms of content and format. We are doing it steadily.

File: skabase/SKALogger/SKALogger.py (hunk 1 above)
Comment (adityadangeska) [reply skabase/SKALogger/SKALogger.py]: Added original exception message. Commit id 35e9e429f5c8d3afd470cdaa81ca011bedcdec83 addresses this.

File: skabase/SKALogger/SKALogger.py (hunk 3 above)
Comment (adityadangeska) [reply skabase/SKALogger/SKALogger.py]: Added original exception message. Commit id 35e9e42 addresses this.

File: skabase/SKALogger/SKALogger.py (hunk 4 above)
Comment (adityadangeska) [reply skabase/SKALogger/SKALogger.py]: Added original exception message. Commit id 35e9e42 addresses this.

File: skabase/SKALogger/SKALogger.py (hunk 2 above)
Comment (adityadangeska) [reply skabase/SKALogger/SKALogger.py]: Corrected. Commit id 35e9e42 addresses this.

File: skabase/SKALogger/SKALogger.py [hunk 7]
```diff
@@ -235,8 +237,11 @@ def SetElementLoggingLevel(self, argin):
                 dev_proxy = DeviceProxy(element_logging_device[i])
                 dev_proxy.elementLoggingLevel = element_logging_level[i]
                 i += 1
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: This increment needs to be removed.

File: skabase/SKALogger/SKALogger.py [hunk 8]
```diff
@@ -272,8 +277,11 @@ def SetStorageLoggingLevel(self, argin):
                 dev_proxy = DeviceProxy(storage_logging_device[i])
                 dev_proxy.storageLoggingLevel = storage_logging_level[i]
                 i += 1
```
Comment (ajoubertza) [inline skabase/SKALogger/SKALogger.py]: Ditto

**Review by ajoubertza — CHANGES_REQUESTED** [review]
Thanks for improvements, but still some bugs.

File: skabase/SKALogger/SKALogger.py (hunk 7 above)
Comment (adityadangeska) [reply skabase/SKALogger/SKALogger.py]: The increment is required as there could be list of devices for which logging level need to be set.
Do you suggest to implement entirely different logic to iterate on the list?

Comment (adityadangeska) [reply skabase/SKALogger/SKALogger.py]: Sorry.. Corrected. Commit #705831b58ecfa5cbf8a2fb7610047c649c603de8 addresses this.

File: skabase/SKALogger/SKALogger.py (hunk 8 above)
Comment (adityadangeska) [reply skabase/SKALogger/SKALogger.py]: Sorry.. Corrected. Commit #705831b58ecfa5cbf8a2fb7610047c649c603de8 addresses this.

**Review by ajoubertza — APPROVED** [review]
Looks good.

**Pull request merged**